2017/Team-Training/OMERO-web/index.html

```diff
@@ -0,0 +1,566 @@
+                        </li>
+                        <li>Composition
+                            <pre><code>{% includes "path/to/template.html" %}</code></pre>
+                        </li>
+                        <li>
+                            OMERO.web: extensive <a target="_blank" href="https://gist.github.com/will-moore/ff3b9d52df40c786701279df4914185c">hierarchy of templates</a>
+                        </li>
+                        <li>
+                            Needs to be simplified (see
+                            <a target="_blank" href="https://trello.com/c/BQyJ0Ay7/47-django-templates-cleanup">Trello card</a>)
+                        </li>
+                    </ul>
+                </section>
+
+
+                <section>
+                    <h2>Sequence of /webclient/ app</h2>
+
+                    <ol>
+                        <li>Load containers.html (includes jQuery and lots of other JavaScript)</li>
+                        <li>jQuery loads other data via AJAX and updates page HTML (DOM):
+                            <ul>
+                                <li>jsTree loads P/D/I S/P as JSON data</li>
+                                <li>jsTree JSON data used to build Dataset thumbnails HTML</li>
+                                <li>Selected items trigger right panel loading HTML</li>
+                                <li>Plates loaded as JSON data</li>
+                                <li>Scripts loaded as JSON data</li>
+                                <li>Groups/Users menu loaded as HTML</li>
+                                <li>etc....</li>
+                            </ul>
+                        </li>
+                    </ol>
+                </section>
+
+
+                <section>
+                    <h2>Browser devtools (Chrome or Firefox)</h2>
+
+                    <ul>
+                        <li>Inspect DOM (right click on Element -> Inspect) <br>
+                            Find elements, IDs, classes etc.
+                            </li>
+                        <li>Edit CSS in the browser</li>
+                        <li>Access JavaScript Console</li>
+                        <li>Inspect Network requests
+                            <ul>
+                                <li>Requested URL and GET/POST</li>
+                                <li>What initialised it?</li>
+                                <li>Data sent and Recieved</li>
+                                <li>Timing</li>
+                                <li>Open GET URLs in new Tab</li>
+                            </ul>
+                        </li>
+                    </ul>
+                </section>
+
+
+                <section>
+                    <h2>jQuery</h2>
+
+                    <ul>
+                        <li><strong>Very</strong> popular library for DOM manipulation, AJAX etc</li>
+                        <li>We use jQuery 1.11.1, 2014 (current version is 3.2.1)</li>
```

Comment:
Is there a need to explain why an old version is used

Reply:
There's no real reason. It's just not been a priority. I can mention that in the training.